packages/flutter_tools/lib/src/service_protocol.dart

```diff
@@ -8,9 +8,13 @@ import 'device.dart';
 
 /// Discover service protocol ports on devices.
 class ServiceProtocolDiscovery {
-  /// [logReader] A [DeviceLogReader] to look for Observatory messages in.
-  ServiceProtocolDiscovery(DeviceLogReader logReader)
-      : _logReader = logReader {
+  static const String kObservatoryService = 'Observatory';
+  static const String kDiagnosticService = 'Diagnostic server';
```

Comment:
Yes - the diagnostic server needs to notify the tools about its chosen port (similar to the observatory).  I'm assuming that disabling features like this will be one of the differences between deployment and development mode.